kernel/src/util/random.rs

```diff
@@ -37,6 +37,14 @@ pub fn init() {
                 tail.copy_from_slice(&src[..n]);
             }
 
+            RNG.call_once(|| SpinLock::new(StdRng::from_seed(seed)));
+        } else if #[cfg(target_arch = "riscv64")] {
+            use rand::SeedableRng;
+            use ostd::arch::boot::DEVICE_TREE;
+
+            let chosen = DEVICE_TREE.get().unwrap().find_node("/chosen").unwrap();
+            let seed = chosen.property("rng-seed").unwrap().value.try_into().unwrap();
+
```

Comment:
Can we let the `if-else` returns `seed` and init the RNG out of the branches? This reduces code redundancy and improves readability.